### PR TITLE
feat: add `math/base/special/cotd`

### LINKED ISSUE #40: Implement `cotd`
Compute the cotangent of `x` where `x` is in degrees. `@stdlib/math/base/special/cotd`.

### PR DESCRIPTION
**Pull request by saisrikardumpeti**
Resolves #40 

## Description

> What is the purpose of this pull request?

This pull request:

-   a
-   b
-   c

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by saisrikardumpeti** [comment]
@Planeshifter should I add checks for when input is `-180` it should return `-Infinity` because I saw in `julia` when passed `-180` it is returning `-Inf`

**Comment by saisrikardumpeti** [comment]
@Planeshifter this PR is ready for review.

File: lib/node_modules/@stdlib/math/base/special/cotd/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# cotd
+
+> Computes the [cotangent][trigonometric-functions] of an angle measured in degrees..
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/README.md]:
```suggestion
> Computes the [cotangent][trigonometric-functions] of an angle measured in degrees.
```

File: lib/node_modules/@stdlib/math/base/special/cotd/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Evaluates the [cotangent][trigonometric-functions] of `x` (in degrees).
+
+```javascript
+var v = cotd( 0.0 );
+// returns Infinity
+
+v = cotd( 60 );
+// returns ~0.58
+
+v = cotd( 90 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/README.md]: Would you mind adding `.0` to the input values where applicable? We include them when the input parameter is for floating-point numbers and is not restricted to integers.

```suggestion
v = cotd( 60.0 );
// returns ~0.58

v = cotd( 90.0 );
```

File: lib/node_modules/@stdlib/math/base/special/cotd/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+
+{{alias}}( x )
+    Computes the cotangent of a degrees.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/docs/repl.txt]:
```suggestion
    Computes the cotangent of an angle measured in degrees.
```

File: lib/node_modules/@stdlib/math/base/special/cotd/docs/repl.txt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        Cotangent.
+
+    Examples
+    --------
+    > var y = {{alias}}( 0.0 )
+    Infinity
+    > y = {{alias}}( 90 )
+    0.0
+    > y = {{alias}}( 60 )
+    ~0.58
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/docs/repl.txt]:
```suggestion
    > y = {{alias}}( 90.0 )
    0.0
    > y = {{alias}}( 60.0 )
    ~0.58
```

File: lib/node_modules/@stdlib/math/base/special/cotd/docs/types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+ * @example
+ * var v = cotd( 0.0 );
+ * // returns Infinity
+ *
+ * @example
+ * var v = cotd( 60 );
+ * // returns ~0.58
+ *
+ * @example
+ * var v = cotd( 90 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/docs/types/index.d.ts]:
```suggestion
 * var v = cotd( 60.0 );
 * // returns ~0.58
 *
 * @example
 * var v = cotd( 90.0 );
```

File: lib/node_modules/@stdlib/math/base/special/cotd/lib/index.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+* // returns Infinity
+*
+* v = cotd( 45 );
+* // returns 1.0
+*
+* v = cotd( 90 );
+* // returns 0.0
+*
+* v = cotd( 60 );
+* // returns ~0.58
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/lib/index.js]:
```suggestion
* v = cotd( 45.0 );
* // returns 1.0
*
* v = cotd( 90.0 );
* // returns 0.0
*
* v = cotd( 60.0 );
* // returns ~0.58
```

File: lib/node_modules/@stdlib/math/base/special/cotd/lib/main.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+* // returns ~0.58
+*
+* @example
+* var v = cotd( NaN );
+* // returns NaN
+*/
+function cotd( x ) {
+	var xRad;
+
+	if ( x === Infinity || x === -Infinity ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/lib/main.js]: Could you please replace this check with usage of `isInfinite` required via `var isInfinite = require( '@stdlib/math/base/assert/is-infinite' );`?

File: lib/node_modules/@stdlib/math/base/special/cotd/lib/main.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+		return NaN;
+	}
+
+	if ( isInteger( ( ( x / 90 ) - 1 ) / 2 ) ) {
+		return 0.0;
+	}
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/lib/main.js]:
```suggestion
	if ( isInteger( ( ( x / 90.0 ) - 1.0 ) / 2.0 ) ) {
		return 0.0;
	}
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Left some small change requests that hopefully will be easy to resolve.

Ready for merging otherwise, I think.

**Comment by saisrikardumpeti** [comment]
@Planeshifter i have made the changes to this PR and others too. It is ready to be merged :rocket:

File: lib/node_modules/@stdlib/math/base/special/cotd/README.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# cotd
+
+> Computes the [cotangent][trigonometric-functions] of an angle measured in degrees.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/README.md]:
```suggestion
> Compute the [cotangent][trigonometric-functions] of an angle measured in degrees.
```

File: lib/node_modules/@stdlib/math/base/special/cotd/package.json [hunk 10]
```diff
@@ -0,0 +1,69 @@
+{
+  "name": "@stdlib/math/base/special/cotd",
+  "version": "0.0.0",
+  "description": "Computes the cotangent of an angle measured in degrees",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cotd/package.json]:
```suggestion
  "description": "Compute the cotangent of an angle measured in degrees",
```

**Review by Planeshifter — APPROVED** [review]
Let's get this one in as well; thanks!

**Comment by kgryte** [comment]
@the-r3aper7 Same comment. Can you tick the box in the OP indicating that you've read the contributing guidelines? Thanks!

**Pull request merged**